src/components/accessories/patientTherapy/patientTherapyTable/PatientTherapyTable.tsx

```diff
@@ -11,21 +14,34 @@ const PatientTherapyTable: FunctionComponent<IOwnProps> = ({
 }) => {
   const [, setUpdate] = useState(false);
   const { t } = useTranslation();
-
-  const label = {
-    SD: "Start date",
-    ED: "End date",
-    medical: t("Medical"),
-    Qty: t("Quantity"),
-    FID: "Frequence In Day",
-    D: t("Duration"),
-    FIP: t("Frequence In Period"),
-  };
+  const therapy = useSelector<IState, ITherapiesState>(
+    (state) => state.therapies
+  );
+  if (therapy.createTherapy.data) {
+    const newRow = {
+      startDate: therapy.createTherapy.data!.startDate!,
+      endDate: therapy.createTherapy.data!.endDate!,
+      medicalId: therapy.createTherapy.data!.medicalId! + "",
+      qty: therapy.createTherapy.data!.qty!,
+      freqInDay: therapy.createTherapy.data!.freqInDay!,
+      freqInPeriod: therapy.createTherapy.data!.freqInPeriod!,
+      unitID: therapy.createTherapy.data!.unitID!,
+      note: therapy.createTherapy.data!.note!,
+    };
+    const found = data.some((el) => el.medicalId === newRow.medicalId);
+    if (!found) data.push(newRow);
+  }
 
   const onDelete = () => {
     console.log("delete");
   };
 
+  const onEdit = () => {
+    console.log("update");
+  };
+
+  const onEView = () => {};
+
```

Comment:
It looks like there's something missing here.

Reply:
I think for the table manipulation the common logic should be defined first. Or may be I am missing it.

Reply:
At this moment we do not yet have a table manipulation common logic, or may be I am missing it